launcher/src/main/java/org/apache/spark/launcher/SparkSubmitCommandBuilder.java

```diff
@@ -138,25 +139,30 @@
 
         case RUN_EXAMPLE:
           isExample = true;
+          appResource = SparkLauncher.NO_RESOURCE;
           submitArgs = args.subList(1, args.size());
       }
 
       this.isExample = isExample;
-      OptionParser parser = new OptionParser(true);
-      parser.parse(submitArgs);
-      this.requiresAppResource = parser.requiresAppResource;
+      if (!submitArgs.isEmpty()) {
```

Comment:
Do you need this check? `parser.isSpecialCommand` is initialized to `false`, so seems like it's already handled.

Reply:
Removed as it leads to the same result.